### DISCUSSION TITLE
FYI jin.ai reader can now be self hosted

### DISCUSSION THREAD

**Discussion by thiswillbeyourgithub** [open-webui/open-webui#5789] [Ideas]
Hi,

For those wondering, jin.ai reader is something that makes it extra easy to turn a URL into LLM friendly markdown output. Including base64 images etc.

Link : https://github.com/jina-ai/reader

Now, even though it was on GitHub, there were some key components missing that made it impossible to self-host until now. 

Someone made a docker container for it, making it easy to self-host the whole thing. Here's the link: https://github.com/intergalacticalvariable/reader

Unfortunately, I'm not able to test it right now, but I thought it was worth mentioning here, because I think it would be a nice addition to an open WebUI Docker Compose setup with a simple tool to send URLs.

**Comment by intergalacticalvariable** [comment]
Unfortunately, image processing is not working for now.

**Reply by thiswillbeyourgithub** [reply]
Is it planned by any chance? Also do you know if it was part of the non opened part of jina reader or is most of the code is in their repo?

**Reply by intergalacticalvariable** [reply]
It cat take screenshots of the page but it does not extracts pictures from website.
```
curl -H "x-respond-with: pageshot" 'https://reader.berlin.cx/https://www.google.com/search?q=beste+bars+in+berlin&start=10'
```
http://reader.berlin.cx/instant-screenshots/pageshot-2a302cbf-fd44-47d5-bdef-5b85aa3e5b5a.png

Visit https://reader.berlin.cx/{any url} for markdown output or curl example from above to take screenshot/pageshot
Example: https://reader.berlin.cx/https://www.google.com/search?q=beste+bars+in+berlin&start=10

Demo is slow bc of sever specs CPU: 1 vCore, RAM: 0.5 GB
